arviz/tests/external_tests/test_data_pyro.py

```diff
@@ -123,3 +162,71 @@ def model_example_multiple_obs(y1=None, y2=None):
         fails = check_multiple_attrs(test_dict, inference_data)
         assert not fails
         assert not hasattr(inference_data.sample_stats, "log_likelihood")
+
+    def test_inference_data_constant_data(self):
+        import pyro.distributions as dist
+        from pyro.infer import MCMC, NUTS
+
+        x1 = 10
+        x2 = 12
+        y1 = torch.randn(10)
+
+        def model_constant_data(x, y1=None):
+            _x = pyro.sample("x", dist.Normal(1, 3))
+            pyro.sample("y1", dist.Normal(x * _x, 1), obs=y1)
+
+        nuts_kernel = NUTS(model_constant_data)
+        mcmc = MCMC(nuts_kernel, num_samples=10)
+        mcmc.run(x=x1, y1=y1)
+        posterior = mcmc.get_samples()
+        posterior_predictive = Predictive(model_constant_data, posterior)(x1)
+        predictions = Predictive(model_constant_data, posterior)(x2)
+        inference_data = from_pyro(
+            mcmc,
+            posterior_predictive=posterior_predictive,
+            predictions=predictions,
+            constant_data={"x1": x1},
+            predictions_constant_data={"x2": x2},
+        )
+        test_dict = {
+            "posterior": ["x"],
+            "posterior_predictive": ["y1"],
+            "sample_stats": ["diverging"],
+            "log_likelihood": ["y1"],
+            "predictions": ["y1"],
+            "observed_data": ["y1"],
+            "constant_data": ["x1"],
+            "predictions_constant_data": ["x2"],
+        }
+        fails = check_multiple_attrs(test_dict, inference_data)
+        assert not fails
+
+    def test_inference_data_dims(self, data, eight_schools_params):
+        inference_data = self.get_inference_data(data, eight_schools_params)
+        test_dict = {
+            "posterior": ["mu", "tau", "eta"],
+            "sample_stats": ["diverging"],
+            "posterior_predictive": ["obs"],
+            "predictions": ["obs"],
+            "prior": ["mu", "tau", "eta"],
+            "prior_predictive": ["obs"],
+        }
+        fails = check_multiple_attrs(test_dict, inference_data)
+        assert not fails
+
+        dims = inference_data.posterior_predictive["obs"].shape[2:]
+        pred_dims = inference_data.predictions["obs"].shape[2:]
+        assert dims == (8, )
+        assert pred_dims == (8, )
+
+    def test_inference_data_num_chains(self, data):
+        posterior_samples = data.obj.get_samples()
+        model = data.obj.kernel.model
+        predictions_data = {"J": 8, "sigma": np.array([5.0, 7.0, 12.0, 4.0, 6.0, 10.0, 3.0, 9.0])}
+        predictions = Predictive(model, posterior_samples)(
+            predictions_data["J"], torch.from_numpy(predictions_data["sigma"]).float()
+        )
+
+        inference_data = from_pyro(predictions=predictions, num_chains=2)
+        nchains = inference_data.predictions["obs"].shape[0]
```

Comment:
`inference_data.predictions.dims["chain"]` should return the length of chain dim